### DISCUSSION TITLE
Feature request: disable FLoC in future qutebrowser releases (Federated Learning of Cohorts)

### DISCUSSION THREAD

**Discussion by twodayporkathon** [qutebrowser/qutebrowser#6437] [Ideas] [answered]
Hi! I'd like to put in a request to disable FLoC in future qutebrowser releases. I'm not sure which version of Chromium first adopted FLoC, but this is where the library is defined in Chromium's source code:

https://chromium.googlesource.com/chromium/src/+/refs/tags/90.0.4430.83/chrome/browser/BUILD.gn

From BUILD.gn:

    123 static_library("browser") {
    124   sources = [
    ...
    325     "component_updater/floc_component_installer.cc",
    326     "component_updater/floc_component_installer.h",
    ...
    514     "federated_learning/floc_eligibility_observer.cc",
    515     "federated_learning/floc_eligibility_observer.h",
    516     "federated_learning/floc_event_logger.cc",
    517     "federated_learning/floc_event_logger.h",
    518     "federated_learning/floc_id_provider.h",
    519     "federated_learning/floc_id_provider_factory.cc",
    520     "federated_learning/floc_id_provider_factory.h",
    521     "federated_learning/floc_id_provider_impl.cc",
    522     "federated_learning/floc_id_provider_impl.h",
    523     "federated_learning/floc_remote_permission_service.cc",
    524     "federated_learning/floc_remote_permission_service.h",
    525     "federated_learning/floc_remote_permission_service_factory.cc",
    526     "federated_learning/floc_remote_permission_service_factory.h",
    ...
    1947  deps = [
    ...
    2093    "//components/federated_learning",
    ...

Don't know when these were merged into Chromium builds. Was qutebrowser v2.2.1 released with FLoC active?

**Comment by twodayporkathon** [comment]
Afterthought: optionally disabling FLoC may be better. I wonder how many web services would break with FLoC disabled two years from now.

**Comment by The-Compiler** [comment, marked as answer]
Taking the freedom to copy-paste an earlier answer from IRC:

12:01 \<kraM\> Hey guys. What are the plans for FLoC? Will qutebrowser protect me from getting "FLoCed"?
12:06 \<The-Compiler\> for context: " Federated Learning of Cohorts or FLoC, Google’s latest targeted advertising experiment"
12:06 \<The-Compiler\> https://www.eff.org/deeplinks/2021/03/googles-floc-terrible-idea
12:06 \<The-Compiler\> kraM: no idea though, to be honest
12:08 \<The-Compiler\> looks like that trial launches in Chromium 89, so will only be relevant with Qt 6.2 in September (if they update to 89 until then, AFAIK they're working on 88 now)
12:09 \<The-Compiler\> but even then, chances are that part of the code isn't even active with QtWebEngine
12:11 \<kraM\> I think FLoC is one of the worst ideas ever and may have huge impact on things way outside of the scope to provide even more money/power to google. And I really think qutebrowser should follow Vivaldi and others to protect people from google.
12:40 \<ninewise\> kraM, if it's avoidable (which is probably will be) it's very unlikely qutebrowser will turn it on by default
12:40 \<ninewise\> there might be a setting to turn it on though, since there's a setting for everything
13:05 \<kraM\> ninewise: I don't think this will be avoidable easily in future chromium versions with one configuration switch. Today you'd have to disable a lot of "features" to "opt-out" in terms of "there is just less data for FLoC to look at". You'd still get your FLoC ID and still data from the browser would be aggregated and sent to websites and google.
13:21 \<nimaje\> qt-webengine will probably remove that stuff or make it configurable
14:12 \<The-Compiler\> kraM: Sure, I guess everyone agrees to that, but there's a gap between what I'd like qutebrowser to do privacy-wise (or expose via options) and what QtWebEngine/Chromium lets it do
14:36 \<kraM\> The-Compiler: That's my main concern. If qb wasn't able to protect users from FLoC, this would mean that qb wouldn't be a viable alternative anymore. (imho)

**Comment by The-Compiler** [comment]
Oh, also let me note that QtWebEngine [has an explicit list of files in `browser/` to include](https://github.com/qt/qtwebengine/blob/v5.15.3-lts/tools/scripts/take_snapshot.py#L69-L96), so unless they actively change that to include `federated_learning/`, that part of the sources won't even be part of QtWebEngine at all.

**Comment by twodayporkathon** [comment]
Till now I didn't know how qutebrowser interacts with the QtWebEngine. Thanks!
